transformers/shared/constants/external.go

```diff
@@ -62,40 +64,58 @@ func GetTransformerContractNames(transformerLabel string) []string {
 	configKey := "exporter." + transformerLabel + ".contracts"
 	contracts := viper.GetStringSlice(configKey)
 	if len(contracts) == 0 {
-		log.Fatalf("No contracts configured for transformer: \"%v\"", transformerLabel)
+		logrus.Fatalf("No contracts configured for transformer: \"%v\"", transformerLabel)
 	}
 	return contracts
 }
 
-// Get the ABI for multiple contracts from config
+// GetABIFromContractsWithMatchingABI gets the ABI for multiple contracts from config
 // Makes sure the ABI matches for all, since a single transformer may run against many contracts.
-func GetContractsABI(contractNames []string) string {
+func GetABIFromContractsWithMatchingABI(contractNames []string) string {
 	initConfig()
 	if len(contractNames) < 1 {
-		log.Fatalf("No contracts to get ABI for")
+		logrus.Fatalf("No contracts to get ABI for")
 	}
-	abi := getContractABI(contractNames[0])
-	for _, contractName := range contractNames[:1] {
-		if abi != getContractABI(contractName) {
-			log.WithField("contracts", contractNames).Fatalf("ABIs not consistent between contracts")
+	contractABI := getContractABI(contractNames[0])
+	parsedABI, parseErr := eth.ParseAbi(contractABI)
+	if parseErr != nil {
+		panic(fmt.Sprintf("unable to parse ABI for %s", contractNames[0]))
+	}
+	for _, contractName := range contractNames[1:] {
+		nextABI := getContractABI(contractName)
+		nextParsedABI, nextParseErr := eth.ParseAbi(nextABI)
+		if nextParseErr != nil {
+			panic(fmt.Sprintf("unable to parse ABI for %s", contractName))
+		}
+		if !parsedABIsAreEqual(parsedABI, nextParsedABI) {
+			panic(fmt.Sprintf("ABIs don't match for contracts: %s", contractNames))
 		}
 	}
-	return abi
+	return contractABI
+}
+
+// GetFirstABI returns the ABI from the first contract in a collection in config
+func GetFirstABI(contractNames []string) string {
```

Comment:
That's now used when generating the [transformer config](https://github.com/makerdao/vdb-mcd-transformers/blob/staging/transformers/shared/config.go) - since we have events that live across contracts with different ABIs (e.g. deal on flap/flip/flop)